.github/workflows/sphinx.yml

```
@@ -21,7 +24,18 @@ jobs:
       run: |
         python -m pip install --upgrade pip
         if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
+    - name: Remove Existing .rst Files (excluding index.rst and modules.rst)
+      run: find docs -maxdepth 1 -type f -name '*.rst' ! -name 'index.rst' ! -name 'modules.rst' -exec rm -f {} \;
+    - name: Generate Sphinx Documentation with sphinx-apidoc
+      run: sphinx-apidoc -o docs .
```

```suggestion
      run: sphinx-apidoc -o docs . tests/
```
Adding `tests/` will stop documentation from being generated for all the tests - which I think would be a good thing. Our testing is not very complex, and IMO does not need Sphinx documentation.

---

That makes sense to me. I'd be curious what others think about that as well though. @allisterakun @ltp23 @PooyaHekmati thoughts?

---

I would agree with Ed about excluding the unit tests. I think all the unit tests I have seen so far are self-descriptive enough; we don't need explicit documentation.

---

Thank you @allisterakun and @Isgotkowitz, I will add Ed's line and make sure it's running as intended before merging.